R/knitr-engine.R

```diff
@@ -118,34 +110,45 @@ eng_python <- function(options) {
   # line index from which source should be emitted
   pending_source_index <- 1
 
+  # whether an error occurred during execution
+  had_error <- FALSE
+
   # actual outputs to be returned to knitr
   outputs <- list()
 
   # synchronize state R -> Python
   eng_python_synchronize_before()
 
+  # determine if we should capture errors (don't capture in rmarkdown::render())
+  capture_errors <- local({
+
+    if (identical(options$error, TRUE))
+      return(TRUE)
+
+    if (!requireNamespace("rmarkdown", quietly = TRUE))
+      return(TRUE)
+
+    rmarkdown <- asNamespace("rmarkdown")
+    context <- rmarkdown$.render_context
```

Comment:
We can use some R options that are set only during knit and/or render as described here: https://github.com/yihui/knitr/issues/926

So for knitr:

```r
isTRUE(getOption('knitr.in.progress'))
```

And for R Markdown, you can check for any one of the following values set in opts_knit:

```r
https://github.com/rstudio/rmarkdown/blob/master/R/render.R#L488-L496
```